src/safeposix/syscalls/net_calls.rs

```diff
@@ -1893,39 +2009,62 @@ impl Cage {
                     }
                 }
 
+                //Gain write access to the socket handle to insert
+                //info regarding the unix connection
                 let newsock_tmp = newsockfd.handle.clone();
                 let mut newsockhandle = newsock_tmp.write();
 
+                //Retrieve the inodenum of the peer socket
                 let pathclone = normpath(convpath(remote_addr.path()), self);
                 if let Some(inodenum) = metawalk(pathclone.as_path()) {
+                    //Insert necessary info about the socket communication
                     newsockhandle.unix_info = Some(UnixSocketInfo {
                         inode: inodenum.clone(),
                         mode: sockhandle.unix_info.as_ref().unwrap().mode,
                         sendpipe: Some(sendpipenumber.clone()),
                         receivepipe: Some(receivepipenumber.clone()),
                     });
+                    //Grab the peer socket inode from the inodetable 
+                    //and increase the refcount by 1
+                    // ** Why is this necessary ?? ** //
```

Comment:
not sure, but from what i saw while working with filesystem calls is that close_syscall() looks at the refcount of the file and proceeds with the removal if it is zero. i guess we don't want the open socket to be removed, that is why we increase its refcount.